packages/app/src/pages/Specs/Runner.vue

```diff
@@ -1,22 +1,66 @@
 <template>
-  <SpecRunnerContainer
-    v-if="query.data.value?.currentProject?.specs"
+  <!--
+    Run Mode is a more minimal UI.
+    It does not render things like the SpecList,
+    Side and Top Nav, etc.
+    It also has no GraphQL dependency.
+  -->
+  <SpecRunnerContainerRunMode
+    v-if="isRunMode"
+    :run-mode-specs="specs"
+  />
+
+  <!--
+    Open Mode is the full Cypress runner UI -
+    including things like the SpecList,
+    Side and Top Nav, Selector Playgroundn etc.
+    It is driven by GraphQL and urql.
+  -->
+  <SpecRunnerContainerOpenMode
+    v-else-if="query.data.value?.currentProject?.specs"
     :gql="query.data.value"
   />
 </template>
 
 <script lang="ts" setup>
 import { gql, useQuery } from '@urql/vue'
 import { SpecPageContainerDocument } from '../../generated/graphql'
-import SpecRunnerContainer from '../../runner/SpecRunnerContainer.vue'
+import { getAutIframeModel } from '../../runner'
+import SpecRunnerContainerOpenMode from '../../runner/SpecRunnerContainerOpenMode.vue'
+import SpecRunnerContainerRunMode from '../../runner/SpecRunnerContainerRunMode.vue'
+import { togglePlayground } from '../../runner/utils'
 
 gql`
 query SpecPageContainer {
   ...SpecRunner
 }
 `
 
-const query = useQuery({ query: SpecPageContainerDocument })
+const isRunMode = window.__CYPRESS_MODE__ === 'run'
+
+// in run mode, we are not using GraphQL or urql
+// for performance - run mode does not need the
+// same level of runner interactivity as open mode.
+// by setting `cache-only` in run mode, urql will not trigger any
+// requests, which is what we want.
+const query = useQuery({
+  query: SpecPageContainerDocument,
+  requestPolicy: isRunMode && window.top === window ? 'cache-only' : 'cache-and-network',
+})
+
+// because we are not using GraphQL in run mode, and we still need
+// way to get the specs, we simply attach them to window when
+// serving the initial HTML.
+// this works fine - we know that during run mode, no new specs will
+// be added or removed.
+const specs = window.__RUN_MODE_SPECS__
+
+// @ts-ignore - this is used for exposing the selector playground in e2e tests
+// TODO: migrate this to true e2e test w/o the hack using Cypress-in-Cypress when
```

Comment:
Can this be done now that Cypress in Cypress is a thing?

Reply:
Should be, gonna migrate a bunch of the specs from packages/runner to here soon as suggested [here](https://github.com/cypress-io/cypress/pull/19625#discussion_r787767103)

That PR might be a lot of code so I'll do it separately, I don't think migrating this spec should block this PR right now. But good idea :+1:

Reply:
Ticket to do this (and others) https://cypress-io.atlassian.net/browse/UNIFY-945